databuilder/databuilder/extractor/pandas_profiling_column_stats_extractor.py

```diff
@@ -0,0 +1,191 @@
+        'p_infinite': ('Infinite values %', lambda x: float(x * 100)),
+        'p_missing': ('Missing values %', lambda x: float(x * 100)),
+        'p_negative': ('Negative values %', lambda x: float(x * 100)),
+        'p_unique': ('Unique values %', lambda x: float(x * 100)),
+        'p_zeros': ('Zeros %', lambda x: float(x * 100)),
+        'range': ('Range', float),
+        'skewness': ('Skewness', float),
+        'std': ('Std. deviation', float),
+        'sum': ('Sum', float),
+        'variance': ('Variance', float)
+        # Stats available in pandas-profiling but are not collected by default and require custom, conscious config..
+        # 'block_alias_char_counts': ('',),
+        # 'block_alias_counts': ('',),
+        # 'block_alias_values': ('',),
+        # 'category_alias_char_counts': ('',),
+        # 'category_alias_counts': ('',),
+        # 'category_alias_values': ('',),
+        # 'character_counts': ('',),
+        # 'cv': ('',),
+        # 'first_rows': ('',),
+        # 'hashable': ('',),
+        # 'histogram': ('',),
+        # 'histogram_frequencies': ('',),
+        # 'histogram_length': ('',),
+        # 'iqr': ('',),
+        # 'length': ('',),
+        # 'mad': ('',),
+        # 'memory_size': ('',),
+        # 'monotonic_decrease': ('Monotonic decrease', bool),
+        # 'monotonic_decrease_strict': ('Strict monotonic decrease', bool),
+        # 'monotonic_increase': ('Monotonic increase', bool),
+        # 'monotonic_increase_strict': ('Strict monotonic increase', bool),
+        # 'n': ('',),
+        # 'n_block_alias': ('',),
+        # 'n_category': ('Categories', int),
+        # 'n_scripts': ('',),
+        # 'ordering': ('',),
+        # 'script_char_counts': ('',),
+        # 'script_counts': ('',),
+        # 'value_counts_index_sorted': ('',),
+        # 'value_counts_without_nan': ('',),
+        # 'word_counts': ('',),
+        # 'type': ('Type', str)
+    }
+
+    PRECISION = 'precision'
+
+    DEFAULT_CONFIG = ConfigFactory.from_dict({STAT_MAPPINGS: DEFAULT_STAT_MAPPINGS, PRECISION: 3})
+
+    def get_scope(self) -> str:
+        return 'extractor.pandas_profiling'
+
+    def init(self, conf: ConfigTree) -> None:
+        self.conf = conf.with_fallback(PandasProfilingColumnStatsExtractor.DEFAULT_CONFIG)
+
+        self._extract_iter = self._get_extract_iter()
+
+    def extract(self) -> Any:
+        try:
+            result = next(self._extract_iter)
+
+            return result
+        except StopIteration:
+            return None
+
+    def _get_extract_iter(self) -> Any:
+        report = self._load_report()
+
+        variables = report.get('variables', dict())
+        report_time = self.parse_date(report.get('analysis', dict()).get('date_start'))
+
+        for column_name, column_stats in variables.items():
+            for _stat_name, stat_value in column_stats.items():
+                stat_spec = self.stat_mappings.get(_stat_name)
+
+                if stat_spec:
```

Comment:
when would `stat_spec` be None? I think it might be nice if we threw if the config was incorrectly typed, just to make the failure happen more clearly/faster

Reply:
the idea is that users might config this publisher with subset of `DEFAULT_STAT_MAPPINGS` - for instance when only selected metrics available in pandas-profiling are meant to make way to amundsen (in total it's around 31 different metrics for numerical values and 17 for text values - it might be too much for some users). in such case stat_spec will be None - _stat_name is present in pandas-profiling but not desired by user to be extracted.

Reply:
The idea is that users might want to extract only subset of metrics available through pandas-profiling (and defined within `DEFAULT_STAT_MAPPINGS`). if users don't want to extract the kurtosis, although it's available in the report file, it would be absent in `self.stat_mappings` and hence - `stat_spec` would be `None`. So it wouldn't be consider a failure but more like filtering out. In total there are 31 unique metrics for numerical values and 17 for text - it could be too much for some users.

Reply:
ohh, I'm sorry, I read this code too fast and though it was iterating through the stat_mappings, not the source mappings. totally makes sense.